File: Python/Easy/145.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        if root is None:
            return []

        answer = []

        # postorder is left right then current
        # left
        answer.extend(self.postorderTraversal(root.left))

        # right
        answer.extend(self.postorderTraversal(root.right))

        # current
        answer.append(root.val)

        return answer

File: Java/Easy/145.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        // base case empty tree
        if (root == null) {
            return new ArrayList<>();
        }

        // postorder is left then right then current
        ArrayList<Integer> answer = new ArrayList<>();

        // left
        answer.addAll(postorderTraversal(root.left));

        // right
        answer.addAll(postorderTraversal(root.right));

        // current
        answer.add(root.val);

        return answer;

    }
}